Packs/ApiModules/Scripts/MicrosoftApiModule/MicrosoftApiModule.py

```diff
@@ -36,33 +36,549 @@ class Resources:
+        ),
+        suffixes=AzureCloudSuffixes(
+            storage_endpoint=suffixes.get('storage_endpoint', defaults.suffixes.storage_endpoint),
+            storage_sync_endpoint=suffixes.get('storage_sync_endpoint', defaults.suffixes.storage_sync_endpoint),
+            keyvault_dns=suffixes.get('keyvault_dns', defaults.suffixes.keyvault_dns),
+            mhsm_dns=suffixes.get('mhsm_dns', defaults.suffixes.mhsm_dns),
+            sql_server_hostname=suffixes.get('sql_server_hostname', defaults.suffixes.sql_server_hostname),
+            mysql_server_endpoint=suffixes.get('mysql_server_endpoint', defaults.suffixes.mysql_server_endpoint),
+            postgresql_server_endpoint=suffixes.get('postgresql_server_endpoint', defaults.suffixes.postgresql_server_endpoint),
+            mariadb_server_endpoint=suffixes.get('mariadb_server_endpoint', defaults.suffixes.mariadb_server_endpoint),
+            azure_datalake_store_file_system_endpoint=suffixes.get(
+                'azure_datalake_store_file_system_endpoint', defaults.suffixes.azure_datalake_store_file_system_endpoint),
+            azure_datalake_analytics_catalog_and_job_endpoint=suffixes.get(
+                'azure_datalake_analytics_catalog_and_job_endpoint',
+                defaults.suffixes.azure_datalake_analytics_catalog_and_job_endpoint),
+            acr_login_server_endpoint=suffixes.get('acr_login_server_endpoint', defaults.suffixes.acr_login_server_endpoint),
+            synapse_analytics_endpoint=suffixes.get('synapse_analytics_endpoint', defaults.suffixes.synapse_analytics_endpoint),
+            attestation_endpoint=suffixes.get('attestation_endpoint', defaults.suffixes.attestation_endpoint),
+        ))
+
+
+def microsoft_defender_for_endpoint_get_base_url(params_endpoint_type, params_url, is_gcc=None):
+    # Backward compatible argument parsing, preserve the url and is_gcc functionality if provided, otherwise use endpoint_type.
+    if is_gcc:  # Backward compatible.
+        endpoint_type = "gcc"
+        params_url = MICROSOFT_DEFENDER_FOR_ENDPOINT_API.get(endpoint_type)
+    elif params_endpoint_type == MICROSOFT_DEFENDER_FOR_ENDPOINT_TYPE_CUSTOM or not params_endpoint_type:
+        # When the integration was configured before our Azure Cloud support, the value will be None.
+        endpoint_type = MICROSOFT_DEFENDER_FOR_ENDPOINT_DEFAULT_ENDPOINT_TYPE  # Default to "com"
+
+        if not params_url:
+            if params_endpoint_type == MICROSOFT_DEFENDER_FOR_ENDPOINT_TYPE_CUSTOM:
+                raise DemistoException("Endpoint type is set to Custom but no URL was provided.")
+            raise DemistoException("Endpoint type is not set and no URL was provided.")
+    else:
+        endpoint_type = MICROSOFT_DEFENDER_FOR_ENDPOINT_TYPE[params_endpoint_type]  # type: ignore[assignment]
+        params_url = params_url or MICROSOFT_DEFENDER_FOR_ENDPOINT_API[endpoint_type]
+    return endpoint_type, params_url
```

Comment:
an alternative way:
to add to MICROSOFT_DEFENDER_FOR_ENDPOINT_TYPE {'Custom': 'com'}

```suggestion
    # Backward compatible argument parsing, preserve the url and is_gcc functionality if provided, otherwise use endpoint_type.
    if is_gcc:  # Backward compatible.
        params_endpoint_type = "US GCC"
    elif params_endpoint_type == MICROSOFT_DEFENDER_FOR_ENDPOINT_TYPE_CUSTOM or not params_endpoint_type:
        # When the integration was configured before our Azure Cloud support, the value will be None.
        if not params_url:
            if params_endpoint_type == MICROSOFT_DEFENDER_FOR_ENDPOINT_TYPE_CUSTOM:
                raise DemistoException("Endpoint type is set to Custom but no URL was provided.")
            raise DemistoException("Endpoint type is not set and no URL was provided.")
    endpoint_type = MICROSOFT_DEFENDER_FOR_ENDPOINT_TYPE.get(params_endpoint_type, 'com')  
    params_url = params_url or MICROSOFT_DEFENDER_FOR_ENDPOINT_API[endpoint_type]
    return endpoint_type, params_url
```